### ISSUE TITLE
New Resample-Syntax leading to cancellation of dimensions

### ISSUE THREAD

**Issue by rpnaut** [pydata/xarray#2356]
#### Example

Starting with the dataset located here: https://swiftbrowser.dkrz.de/public/dkrz_c0725fe8741c474b97f291aac57f268f/GregorMoeller/,
I want to calculate monthly sums of precipitation for each gridpoint in the daily data:

```
In [39]: data = array.open_dataset("eObs_gridded_0.22deg_rot_v14.0.TOT_PREC.1950-2016.nc_CutParamTimeUnitCor_FinalEvalGrid")
In [40]: data
Out[13]: 
<xarray.Dataset>
Dimensions:       (rlat: 136, rlon: 144, time: 153)
Coordinates:
  * rlon          (rlon) float32 -22.6 -22.38 -22.16 -21.94 -21.72 -21.5 ...
  * rlat          (rlat) float32 -12.54 -12.32 -12.1 -11.88 -11.66 -11.44 ...
  * time          (time) datetime64[ns] 2006-05-01T12:00:00 ...
Data variables:
    rotated_pole  int32 ...
    TOT_PREC      (time, rlat, rlon) float32 ...
Attributes:
    CDI:                       Climate Data Interface version 1.8.0 (http://m...
    Conventions:               CF-1.6
    history:                   Thu Jun 14 12:34:59 2018: cdo -O -s -P 4 remap...
    CDO:                       Climate Data Operators version 1.8.0 (http://m...
    cdo_openmp_thread_number:  4

In [41]: datamonth = data["TOT_PREC"].resample(time="M").sum()
In [42]: datamonth
Out[42]: 
<xarray.DataArray 'TOT_PREC' (time: 5)>
array([ 551833.25   ,  465640.09375,  328445.90625,  836892.1875 ,  503601.5    ], dtype=float32)
Coordinates:
  time     (time) datetime64[ns] 2006-05-31 2006-06-30 2006-07-31 ...
```

#### Problem description

The problem is that the dimensions 'rlon' and 'rlat' and the corresponding coordinates have not survived the resample process. Only the time is present in the result. 

#### Expected Output

I expect to have the spatial dimensions still in the output of monthly sums. The surprise is, that this is the case using the old syntax:
```
In [41]: datamonth = data["TOT_PREC"].resample(dim="time",freq="M",how="sum")
/usr/bin/ipython3:1: FutureWarning: 
.resample() has been modified to defer calculations. Instead of passing 'dim' and how="sum", instead consider using .resample(time="M").sum('time') 
  #!/usr/bin/env python3

In [42]: datamonth
Out[42]: 
<xarray.DataArray 'TOT_PREC' (time: 5, rlat: 136, rlon: 144)>
array([[[  0.      ,   0.      , ...,   0.      ,   0.      ],
        [  0.      ,   0.      , ...,   0.      ,   0.      ],
        ..., 
        [  0.      ,   0.      , ...,  44.900028,  41.400024],
        [  0.      ,   0.      , ...,  49.10001 ,  46.5     ]]], dtype=float32)
Coordinates:
  * time     (time) datetime64[ns] 2006-05-31 2006-06-30 2006-07-31 ...
  * rlon     (rlon) float32 -22.6 -22.38 -22.16 -21.94 -21.72 -21.5 -21.28 ...
  * rlat     (rlat) float32 -12.54 -12.32 -12.1 -11.88 -11.66 -11.44 -11.22 ...

```

What is wrong here? 

And maybe I can also ask the question why the new syntax did not consider use cases with high complex scripting? I do not like to use in my programs a hardcoded dimension name, i.e. `time=${freq}` instead of `dim=${dim}; freq=${freq}`.

**Comment by dcherian** [comment]
`datamonth = data["TOT_PREC"].resample(time="M").sum(dim='time')` should do what you want.

**Comment by rpnaut** [comment]
Thank you @dcherian . Do you think, that two times giving the dimension time as argument is useful?

**OR MAYBE i understand everything wrong:**
Is the argument time='M' only mean to be freqency='M'? And the name for the time dimension is now given by the argument "dim"? 
Or let me ask the question different: what would be the syntax of your command, if the time dimension has the name 'TIMES'?

**Comment by dcherian** [comment]
The repeated dimension follows pandas syntax. It's nice because the syntax is similar to the usual reduction `DataArray.sum()`.

`.resample(TIMES='M').sum(dim='TIMES')` should work as long as `TIMES` is datetime64.

**Comment by dcherian** [comment]
Oh sorry, i was mistaken. Looks like pandas does not require the repeated dimension. https://pandas.pydata.org/pandas-docs/stable/timeseries.html#resampling

The current API works like `groupby` but i don't think that was the original intent (#1269 #1272).

**Comment by fujiisoup** [comment]
Thanks, @dcherian.
It looks the original API was designed mainly for 1d arrays, and documentation does not describe clearly how to apply them to multi-dimensional arrays.

I split this issue into two, #2362 and #2363.

**Comment by dcherian** [comment]
Fixed in 0.13.0
